### DISCUSSION TITLE
How do I get the velocity vector of a RigidBody ?

### DISCUSSION THREAD

**Discussion by cark** [jcornaz/heron#248] [Q&A] [answered]
Probably a newbie question but I really couldn't find an answer by myself:

I need the velocity vector of my player units so that ranged enemies can better aim their shots.
The movement of those units is currently in the hands of Heron as I only apply accelerations to these.
How can I access the current simulated velocity of one of these RigidBodies ?

**Comment by jcornaz** [comment, marked as answer]
the [`Velocity`](https://docs.rs/heron/3.0.0/heron/struct.Velocity.html) component (if inserted) should contain the RigidBody Velocity

**Reply by cark** [reply]
Ah indeed. I expected this to only be for setting the Velocity and now verified that it's for reading it as well.
Thank you !

**Reply by cark** [reply]
Might be a good idea to update the docs to reflect this. I've been looking at the Velocity page and didn't consider it at all.

**Reply by jcornaz** [reply]
Yeah, we can improve the doc indeed.
